### PR TITLE
Update the clustering tutorial

### PR DESCRIPTION
**Pull request by WireBaron**
This change updates the clustering tutorial to reflect the current
APIs and address some common pitfalls that might be seen trying to
update a clean postgres install.

### CODE REVIEW

File: tutorials/clustering.md [hunk 1]
```diff
@@ -13,17 +12,25 @@ In this tutorial, we will show how you can create a distributed hypertable withi
...
-Prerequisites:
-- 3 instances of TimescaleDB installed (private BETA branch only)
-- A superuser set up on all 3 instances of TimescaleDB
-- Network connectivity between the 3 TimescaleDB instances
-- Must be on PG11
+It is assumed before starting this tutorial you will have downloaded and installed the
+TimescaleDB multinode beta package on three networked machines running Postgres 11. It's
+also assumed you'll have access to some superuser role common to these three machines.
+
+Beyond this, please ensure that you've followed the following steps. Some or most of
```
Comment (solugebefola) [inline tutorials/clustering.md]:
```suggestion
Beyond this, please ensure that you've completed the following steps. Some or most of
```

File: tutorials/clustering.md [hunk 2]
```diff
@@ -13,17 +12,25 @@ In this tutorial, we will show how you can create a distributed hypertable withi
...
+these may not be necessary depending on how you initially set up your instances (the 
+docker image should be have most of this already configured):
+- Make sure you've enabled 'max_prepared_transactions' on all the instances (set this to 100 if you're unsure), this is found in `postgresql.conf`
```
Comment (solugebefola) [inline tutorials/clustering.md]:
```suggestion
- Make sure you've enabled 'max_prepared_transactions' on all the instances (set this to 100 if you're unsure), this parameter is located in `postgresql.conf`
```

File: tutorials/clustering.md [hunk 3]
```diff
@@ -13,17 +12,25 @@ In this tutorial, we will show how you can create a distributed hypertable withi
...
-- A superuser set up on all 3 instances of TimescaleDB
-- Network connectivity between the 3 TimescaleDB instances
-- Must be on PG11
+It is assumed before starting this tutorial you will have downloaded and installed the
+TimescaleDB multinode beta package on three networked machines running Postgres 11. It's
+also assumed you'll have access to some superuser role common to these three machines.
+
+Beyond this, please ensure that you've completed the following steps. Some or most of
+these may not be necessary depending on how you initially set up your instances (the 
+docker image should be have most of this already configured):
```
Comment (k-rus) [inline tutorials/clustering.md]: None of the items are configured in the image. The first one (max_prepared_transactions) can be done, but not yet. Thus I suggest to remove the text in parenthesis for now.

File: tutorials/clustering.md [hunk 4]
```diff
@@ -13,17 +12,25 @@ In this tutorial, we will show how you can create a distributed hypertable withi
...
+After making any of these changes, please restart your postgres instance to ensure the
+new settings take effect.
 
 ## Define the topology for your distributed database
 
 Now that you have 3 running instances of TimescaleDB, you can go ahead and set up your
-distributed database. Connect to the instance that you would like to act as the access node,
-and create a new database there:
+distributed database. Connect to the instance that you would like to act as the access 
+node using your superuser role, and create a new database there:
```
Comment (k-rus) [inline tutorials/clustering.md]: May be add a superuser to the connection command below, e.g.:
```
\c multinode superuser
```

File: tutorials/clustering.md [hunk 5]
```diff
@@ -13,17 +12,25 @@ In this tutorial, we will show how you can create a distributed hypertable withi
 a distributed database. We will set up the distributed database with one primary access node
 and two data nodes across three distinct TimescaleDB instances.
 
-Prerequisites:
-- 3 instances of TimescaleDB installed (private BETA branch only)
-- A superuser set up on all 3 instances of TimescaleDB
-- Network connectivity between the 3 TimescaleDB instances
-- Must be on PG11
+It is assumed before starting this tutorial you will have downloaded and installed the
+TimescaleDB multinode beta package on three networked machines running Postgres 11. It's
```
Comment (erimatnor) [inline tutorials/clustering.md]: FWIW, I'd like to move away from making "multinode" sound as something different. Let's just explicitly state the prerequisites here.

For instance:

```
The prerequisites for this tutorial are:

- TimescaleDB version 2.0 (beta) or greater.
- PostgreSQL 11
- Three networked PostgreSQL instances with a common superuser role
```

File: tutorials/clustering.md [hunk 6]
```diff
@@ -13,17 +12,25 @@ In this tutorial, we will show how you can create a distributed hypertable withi
...
+these may not be necessary depending on how you initially set up your instances (the 
+docker image should be have most of this already configured):
+- Make sure you've enabled 'max_prepared_transactions' on all the instances (set this to 100 if you're unsure), this parameter is located in `postgresql.conf`
+- Make sure the data nodes are properly listening for network traffic. `postgresql.conf` should have a `listen_addresses` value including the access node's IP. If the field is missing or `localhost`, change it to the needed IP, or `"*"`
+- The superuser role must have an entry in the `pg_hba.conf` file. This should look something like: ```host    all    <superuser role>    <connection source>    trust``` The connection source should be the `<access node's IP>/32` or some encompasing range, or `samenet` if the instances are on the same subnet. For using authentication mechanism's other than `trust`, please see the [authentication guide][data-node-authentication]. Note that if the instance is specifying something other than `pg_hba.conf` in the `hba_file` field of `postgresql.conf`, then that is the file which must have the above line.
+
```
Comment (erimatnor) [inline tutorials/clustering.md]: Not related to this PR, but after reading this, I really think we need to do something to make sure that this is much easier out-of-the box.

File: tutorials/clustering.md [hunk 7]
```diff
@@ -13,17 +12,25 @@ In this tutorial, we will show how you can create a distributed hypertable withi
...
+After making any of these changes, please restart your postgres instance to ensure the
+new settings take effect.
 
 ## Define the topology for your distributed database
 
 Now that you have 3 running instances of TimescaleDB, you can go ahead and set up your
-distributed database. Connect to the instance that you would like to act as the access node,
-and create a new database there:
+distributed database. Connect to the instance that you would like to act as the access
```
Comment (erimatnor) [inline tutorials/clustering.md]: Should probably state somewhere that the data nodes should not have a database on them already.

File: tutorials/clustering.md (hunk 3 above)
Comment (mkindahl) [inline tutorials/clustering.md]: Mentioning the docker image will implicitly add a requirement to use docker (no, it is not necessary, but it will be perceived as the recommended way.) Suggest to remove the comment.

File: tutorials/clustering.md [hunk 8]
```diff
@@ -13,17 +12,25 @@ In this tutorial, we will show how you can create a distributed hypertable withi
...
-- Network connectivity between the 3 TimescaleDB instances
-- Must be on PG11
+It is assumed before starting this tutorial you will have downloaded and installed the
+TimescaleDB multinode beta package on three networked machines running Postgres 11. It's
+also assumed you'll have access to some superuser role common to these three machines.
+
+Beyond this, please ensure that you've completed the following steps. Some or most of
+these may not be necessary depending on how you initially set up your instances (the 
+docker image should be have most of this already configured):
+- Make sure you've enabled 'max_prepared_transactions' on all the instances (set this to 100 if you're unsure), this parameter is located in `postgresql.conf`
```
Comment (mkindahl) [inline tutorials/clustering.md]: Unsure about what? Suggest to write "for example 100" and not in parentheses but rather commas.

File: tutorials/clustering.md [hunk 9]
```diff
@@ -13,17 +12,25 @@ In this tutorial, we will show how you can create a distributed hypertable withi
...
+- Make sure the data nodes are properly listening for network traffic. `postgresql.conf` should have a `listen_addresses` value including the access node's IP. If the field is missing or `localhost`, change it to the needed IP, or `"*"`
```
Comment (mkindahl) [inline tutorials/clustering.md]: This is a little vague on what to set it to. Suggest to write it as "if the IP address of the access node is A.B.C.D, then set listen_address should be set to..."

File: tutorials/clustering.md [hunk 10]
```diff
@@ -13,17 +12,25 @@ In this tutorial, we will show how you can create a distributed hypertable withi
...
+- The superuser role must have an entry in the `pg_hba.conf` file. This should look something like: ```host    all    <superuser role>    <connection source>    trust``` The connection source should be the `<access node's IP>/32` or some encompasing range, or `samenet` if the instances are on the same subnet. For using authentication mechanism's other than `trust`, please see the [authentication guide][data-node-authentication]. Note that if the instance is specifying something other than `pg_hba.conf` in the `hba_file` field of `postgresql.conf`, then that is the file which must have the above line.
```
Comment (mkindahl) [inline tutorials/clustering.md]: This is a little too many options. Suggest to pick a standard one and go for that, even if it is not entirely accurate. Users that have "strange" setups will know how to handle that.

File: tutorials/clustering.md [hunk 11]
```diff
@@ -13,17 +12,25 @@ In this tutorial, we will show how you can create a distributed hypertable withi
...
+
+After making any of these changes, please restart your postgres instance to ensure the
```
Comment (mkindahl) [inline tutorials/clustering.md]: "Postgres"

File: tutorials/clustering.md (hunk 5 above)
Comment (mkindahl) [reply tutorials/clustering.md]: I would also suggest giving these three instances (domain) names and use that in the description below so that you instead of saying "connect to the instance that you want to be an access node", you can write "connect to access.example.com".

Comment (mkindahl) [reply tutorials/clustering.md]: Name the superuser role here as well---that is, 'postgres'---and use it in the description below.

File: tutorials/clustering.md [hunk 12]
```diff
@@ -35,26 +42,15 @@ Now, you can tell your access node that currently contains the multinode databas
...
-bootstrap_password=>'<superuser_password>');
+SELECT add_data_node('dn1', host=>'<data node 1 address>');
 
-SELECT add_data_node('dn2',host=>'<address>', if_not_exists=>'true',
-password=>'<remote_password>', bootstrap_user=>'<superuser>',
-bootstrap_password=>'<superuser_password>');
+SELECT add_data_node('dn2', host=>'<data node 2 address>');
 ```
 
 Let's go through exactly what the `add_data_node` command is doing. The first
```
Comment (mkindahl) [inline tutorials/clustering.md]: No need to "prepare the user" here, just start describing. Suggest to remove first sentence (yes, it is not part of this patch, but anyway).

**Comment by erimatnor** [comment]
Note that I did another PR to update he examples in this tutorial so there's some conflicts to resolve.

File: tutorials/clustering.md (hunk 5 above)
Comment (WireBaron) [reply tutorials/clustering.md]: I've incorporated all of these comments.  I actually lifted the term "TimescaleDB multinode beta" from the instructions Jeff put together, though it might be a bit late to change that now.

File: tutorials/clustering.md (hunk 3 above)
Comment (WireBaron) [reply tutorials/clustering.md]: Good to know, removed the comment.

Comment (WireBaron) [reply tutorials/clustering.md]: removed

File: tutorials/clustering.md (hunk 8 above)
Comment (WireBaron) [reply tutorials/clustering.md]: Yes, that felt clunky when I wrote it and I meant to come back and clean it up.  Still looked a bit funky as you suggested, so I added it as a second sentence here.

File: tutorials/clustering.md (hunk 9 above)
Comment (WireBaron) [reply tutorials/clustering.md]: Cleaned this up.

File: tutorials/clustering.md (hunk 6 above)
Comment (WireBaron) [reply tutorials/clustering.md]: I did try to document everything I stumbled over trying to write the tutorial.

File: tutorials/clustering.md (hunk 11 above)
Comment (WireBaron) [reply tutorials/clustering.md]: Fixed

File: tutorials/clustering.md (hunk 12 above)
Comment (WireBaron) [reply tutorials/clustering.md]: done

File: tutorials/clustering.md (hunk 4 above)
Comment (WireBaron) [reply tutorials/clustering.md]: done

File: tutorials/clustering.md (hunk 7 above)
Comment (WireBaron) [reply tutorials/clustering.md]: Added a sentence to that effect

File: tutorials/clustering.md (hunk 10 above)
Comment (WireBaron) [reply tutorials/clustering.md]: Removed the `samenet` suggestion as well as the last sentence.  I do want to include both IP and hostname though.

**Comment by erimatnor** [comment]
Needs a rebase.

File: tutorials/clustering.md [hunk 13]
```diff
@@ -13,48 +12,53 @@ In this tutorial, we will show how you can create a distributed hypertable withi
...
-- Must be on PG11
+It is assumed before starting this tutorial you will have three networked machines set up with:
+- TimescaleDB version 2.0 (beta) or greater
+- PostgresSQL 11
+
+For the sake of this tutorial we will refer to these machines as
+access.example.com, data1.example.com, data2.example.com. You will also
+need a common superuser role on each postgres instance. This tutorial
+will assume the `postgres` role for this, but you may substitute any
+role common to all machines with superuser permission.
```
Comment (mkindahl) [inline tutorials/clustering.md]: Perfect, but I suggest to put backquotes around the domain names.

File: tutorials/clustering.md [hunk 14]
```diff
@@ -13,48 +12,53 @@ In this tutorial, we will show how you can create a distributed hypertable withi
...
+
+Beyond this, please ensure that you've completed the following steps.
+Some or most of these may not be necessary depending on how you
+initially set up your instances:
+- Make sure you've enabled 'max_prepared_transactions' on all the instances, this parameter is located in `postgresql.conf`. We recommend a setting of at least 100.
```
Comment (mkindahl) [inline tutorials/clustering.md]: Backquotes around `max_prepared_transaction`?

File: tutorials/clustering.md [hunk 15]
```diff
@@ -13,48 +12,53 @@ In this tutorial, we will show how you can create a distributed hypertable withi
...
+new settings take effect.
 
 ## Define the topology for your distributed database
 
-Now that you have 3 running instances of TimescaleDB, you can go ahead and set up your
-distributed database. Connect to the instance that you would like to act as the access node,
-and create a new database there:
+Now that you have 3 running instances of TimescaleDB, you can go ahead
+and set up your distributed database. Connect to `access.example.com`
+using your superuser role, and create a new database there. It's
```
Comment (mkindahl) [inline tutorials/clustering.md]: Suggestion: "as `postgres`"

File: tutorials/clustering.md [hunk 16]
```diff
@@ -13,48 +12,53 @@ In this tutorial, we will show how you can create a distributed hypertable withi
...
+important that none of the instances already have a `multinode`
```
Comment (mkindahl) [inline tutorials/clustering.md]: Not very clear if the `multinode` is because there is a hard-coded limitation that disallow `multinode` or if it is the "new database" that you're referring to. Suggest "... and create a new database `multinode`. It's ...".

File: tutorials/clustering.md [hunk 17]
```diff
@@ -13,48 +12,53 @@ In this tutorial, we will show how you can create a distributed hypertable withi
...
 
 ```sql
 CREATE DATABASE multinode;
-\c multinode
+\c multinode postgres
 CREATE EXTENSION timescaledb;
 ```
 
 Now, you can tell your access node that currently contains the multinode database to add
 data nodes in order to store distributed hypertables:
```
Comment (mkindahl) [inline tutorials/clustering.md]: Suggestion: "You can now add the data nodes to the access node."

File: tutorials/clustering.md [hunk 18]
```diff
@@ -13,48 +12,53 @@ In this tutorial, we will show how you can create a distributed hypertable withi
...
-data nodes when running commands on the access node. The `host` parameter is
-what the access node will use to actually access the remote data node. The
-`if_not_exists` parameter will bootstrap a database on the remote data node if
-an associated database does not yet exist. The `password` parameter is used each
-time the access node connects to a data node once the data node is created. This
-`password` is associated with the current user of the system. The
-`bootstrap_user` and `bootstrap_password` parameters are used when a data node is
-initially logged into, and needs to be a superuser. If you're already logged in as a
-superuser, the `bootstrap_user` and `bootstrap_password` parameters are optional.
+The first argument represents the local name of a data node that you can
```
Comment (mkindahl) [inline tutorials/clustering.md]: Suggestion: Replace first sentence with "The first argument is the local name of the data node."

File: tutorials/clustering.md [hunk 19]
```diff
@@ -13,48 +12,53 @@ In this tutorial, we will show how you can create a distributed hypertable withi
...
+use to refer to data nodes when running commands on the access node. The
+`host` parameter is the IP or DNS the access node will use to actually
```
Comment (mkindahl) [inline tutorials/clustering.md]: Suggest "the IP or DNS" ==> "the address of"

File: tutorials/clustering.md [hunk 20]
```diff
@@ -92,7 +96,12 @@ how many chunks it holds by running the following:
 
 ```sql
 SELECT * FROM timescaledb_information.hypertable;
+```
 
+You can also see the specifics of how a particular hypertable is 
+distributed.
```
Comment (mkindahl) [inline tutorials/clustering.md]: Suggestion: "You can also see how the data of a distributed hypertable is distributed on the data nodes by using the `hypertable_data_node_relation_size` function."

File: tutorials/clustering.md (hunk 13 above)
Comment (WireBaron) [reply tutorials/clustering.md]: done

Comment (WireBaron) [reply tutorials/clustering.md]: done

File: tutorials/clustering.md (hunk 15 above)
Comment (WireBaron) [reply tutorials/clustering.md]: done

File: tutorials/clustering.md (hunk 16 above)
Comment (WireBaron) [reply tutorials/clustering.md]: done

File: tutorials/clustering.md (hunk 17 above)
Comment (WireBaron) [reply tutorials/clustering.md]: yeah, that seems a lot cleaner :)

File: tutorials/clustering.md (hunk 18 above)
Comment (WireBaron) [reply tutorials/clustering.md]: Went with "The first argument is the local name of the data node on the access node."

File: tutorials/clustering.md (hunk 19 above)
Comment (WireBaron) [reply tutorials/clustering.md]: went with "IP or DNS address of the data node."  I wanted to make it clear how to do this if the user is doing everything by IP.

File: tutorials/clustering.md (hunk 20 above)
Comment (WireBaron) [reply tutorials/clustering.md]: done

File: tutorials/clustering.md (hunk 14 above)
Comment (WireBaron) [reply tutorials/clustering.md]: good catch

File: tutorials/clustering.md (hunk 18 above)
Comment (mkindahl) [reply tutorials/clustering.md]: I think the mention of the access node is redundant since the command is executed on the access node, but this is better than the first version.

File: tutorials/clustering.md (hunk 19 above)
Comment (mkindahl) [reply tutorials/clustering.md]: It is not very clear what a DNS address is. Is it the address of the DNS server or something else? I think you should call it domain name.

File: tutorials/clustering.md [hunk 21]
```diff
@@ -13,48 +12,50 @@ In this tutorial, we will show how you can create a distributed hypertable withi
...
-what the access node will use to actually access the remote data node. The
-`if_not_exists` parameter will bootstrap a database on the remote data node if
-an associated database does not yet exist. The `password` parameter is used each
-time the access node connects to a data node once the data node is created. This
-`password` is associated with the current user of the system. The
-`bootstrap_user` and `bootstrap_password` parameters are used when a data node is
-initially logged into, and needs to be a superuser. If you're already logged in as a
-superuser, the `bootstrap_user` and `bootstrap_password` parameters are optional.
+The first argument is the local name of a data node on the access node. The
+`host` parameter is the IP or DNS address of the data node.
```
Comment (mkindahl) [inline tutorials/clustering.md]: It is not very clear what a DNS address is, and AFAICT, it is not a common term. Is it the address of the DNS server or something else? (If I search for "dns address" I get descriptions of DNS servers.) I suggest that you call it domain name.

Comment (WireBaron) [reply tutorials/clustering.md]: Changed this to "hostname or IP address."
